src/main/java/flash/npcmod/entity/NpcEntity.java

```diff
@@ -249,8 +250,7 @@ public static NpcEntity fromJson(Level level, JsonObject jsonObject) {
         NpcEntity npcEntity = EntityInit.NPC_ENTITY.get().create(level);
         assert npcEntity != null;
         if (jsonObject.has("currentBehavior"))
-            npcEntity.setCurrentBehavior(
-                    Behavior.fromJSONObject(jsonObject.getAsJsonObject("currentBehavior")).toCompoundTag());
+            npcEntity.setCurrentBehavior(Behavior.fromJSONObject(jsonObject.getAsJsonObject("currentBehavior")));
         npcEntity.setCustomNameVisible(jsonObject.get("nameVisibility").getAsBoolean());
         npcEntity.setDialogue(jsonObject.get("dialogue").getAsString());
         npcEntity.setCustomName(new TextComponent(jsonObject.get("name").getAsString()));
```

Comment:
Again, null checks. Though I'm pretty sure I fixed this in the renderer branch already, let's just merge and find out after you fix the other issues

```
java.lang.NullPointerException: Cannot invoke "com.google.gson.JsonElement.getAsInt()" because the return value of "com.google.gson.JsonObject.get(String)" is null
	at flash.npcmod.entity.NpcEntity.fromJson(NpcEntity.java:257) ~[%2384!/:?] {re:classloading}
	at flash.npcmod.client.gui.screen.SavedNpcsScreen$NpcInfo.fromJson(SavedNpcsScreen.java:198) ~[%2384!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[?:?] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?] {}
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?] {}
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?] {}
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?] {}
	at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?] {}
	at flash.npcmod.client.gui.screen.SavedNpcsScreen$NpcList.<init>(SavedNpcsScreen.java:211) ~[%2384!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at flash.npcmod.client.gui.screen.SavedNpcsScreen.init(SavedNpcsScreen.java:57) ~[%2384!/:?] {re:classloading}
	at net.minecraft.client.gui.screens.Screen.init(Screen.java:423) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.setScreen(Minecraft.java:919) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at flash.npcmod.ClientProxy.openScreen(ClientProxy.java:60) ~[%2384!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at flash.npcmod.network.packets.server.SOpenScreen.lambda$handle$0(SOpenScreen.java:36) ~[%2384!/:?] {re:classloading}
	at net.minecraftforge.network.NetworkEvent$Context.enqueueWork(NetworkEvent.java:213) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2378%2385!/:?] {re:classloading}
	at flash.npcmod.network.packets.server.SOpenScreen.handle(SOpenScreen.java:35) ~[%2384!/:?] {re:classloading}
	at net.minecraftforge.network.simple.IndexedMessageCodec.lambda$tryDecode$3(IndexedMessageCodec.java:115) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2378%2385!/:?] {re:classloading}
	at java.util.Optional.ifPresent(Optional.java:178) ~[?:?] {}
	at net.minecraftforge.network.simple.IndexedMessageCodec.tryDecode(IndexedMessageCodec.java:115) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2378%2385!/:?] {re:classloading}
	at net.minecraftforge.network.simple.IndexedMessageCodec.consume(IndexedMessageCodec.java:153) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2378%2385!/:?] {re:classloading}
	at net.minecraftforge.network.simple.SimpleChannel.networkEventListener(SimpleChannel.java:72) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2378%2385!/:?] {re:classloading}
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-5.0.7.jar%2310!/:?] {}
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-5.0.7.jar%2310!/:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.7.jar%2310!/:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.7.jar%2310!/:?] {}
	at net.minecraftforge.network.NetworkInstance.dispatch(NetworkInstance.java:68) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2378%2385!/:?] {re:classloading}
	at net.minecraftforge.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:78) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2378%2385!/:?] {re:classloading}
	at java.util.Optional.map(Optional.java:260) ~[?:?] {}
	at net.minecraftforge.network.NetworkHooks.onCustomPayload(NetworkHooks.java:78) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2378%2385!/:?] {re:classloading}
	at net.minecraft.client.multiplayer.ClientPacketListener.handleCustomPayload(ClientPacketListener.java:1824) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket.handle(ClientboundCustomPayloadPacket.java:57) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading}
	at net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket.handle(ClientboundCustomPayloadPacket.java:7) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading}
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:22) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading}
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:143) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:22) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading}
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:116) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:104) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.client.Minecraft.runTick(Minecraft.java:1014) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.run(Minecraft.java:664) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:205) ~[forge-1.18.2-40.1.68_mapped_official_1.18.2-recomp.jar%2379!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.fml.loading.targets.ForgeClientUserdevLaunchHandler.lambda$launchService$0(ForgeClientUserdevLaunchHandler.java:24) ~[fmlloader-1.18.2-40.1.68.jar%230!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-9.1.3.jar%2311!/:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:149) [bootstraplauncher-1.0.0.jar:?] {}